numba/tests/test_dyn_array.py

```diff
@@ -509,6 +509,20 @@ def pyfunc(x):
 
 class ConstructorBaseTest(NrtRefCtTest):
 
+    def test_check_zeros_prototype(self):
+        def func():
+            return np.zeros((2, 2), dtype=np.float)
+        cfunc = nrtjit(func)
+        expected = np.zeros((2, 2), dtype=np.float)
+        result = cfunc()
+        np.testing.assert_equal(result, expected)
+        # errors
+        with self.assertRaises(TypingError) as cm:
```

Comment:
I'm getting another error raised from a different place here

Reply:
> I'm getting another error raised from a different place here.

The Numba compiler catches all errors arising from compilation and rewrites them to include contextual information (it prints the source code, points at the line with the problem, marks up the output, trims the backtraces etc). I've just pulled down this patch and triggered the error, what is present is correct.

> I'm also having some trouble with finding the correct place to put the tests,

Putting this test here seems good as it'll get inherited by the tests for `zeros`, `ones` and `empty`.

> I'd really appreciate some help here

Hopefully the above gets you started.

> sorry for some of the weird commits again because of trouble pushing/pulling to/from remote.

No problem, we can fix up the patches if necessary.